src/avk.cpp

```diff
@@ -4923,24 +4923,35 @@ namespace avk
 		result.mPipelineLayoutCreateInfo = vk::PipelineLayoutCreateInfo{}
 			.setSetLayoutCount(static_cast<uint32_t>(descriptorSetLayoutHandles.size()))
 			.setPSetLayouts(descriptorSetLayoutHandles.data())
-			.setPushConstantRangeCount(static_cast<uint32_t>(result.mPushConstantRanges.size())) 
+			.setPushConstantRangeCount(static_cast<uint32_t>(result.mPushConstantRanges.size()))
 			.setPPushConstantRanges(result.mPushConstantRanges.data());
 
 		// 15. Maybe alter the config?!
 		if (aAlterConfigBeforeCreation) {
 			aAlterConfigBeforeCreation(result);
 		}
-		
+
 		rewire_config_and_create_graphics_pipeline(result);
 		return result;
 	}
+
+	renderpass root::replace_render_pass_for_pipeline(graphics_pipeline& aPipeline, renderpass aNewRenderPass)
+	{
+		if (aPipeline->mRenderPass.is_shared_ownership_enabled()) {
+			aNewRenderPass.enable_shared_ownership();
+		}
+
+		std::swap(aPipeline->mRenderPass, aNewRenderPass);
+
+		return aNewRenderPass; // now old render pass

```

Comment:
Maybe it would be better to also implement the `swap` explicitly here for the sake of clarity. In terms of functionality it should be fine.

Reply:
good point. done.